agent/engine/daemonmanager/daemon_manager_linux.go

```diff
@@ -62,6 +61,10 @@ func NewDaemonManager(manageddaemon *md.ManagedDaemon) DaemonManager {
 	return &daemonManager{managedDaemon: manageddaemon}
 }
 
+func (dm *daemonManager) GetManagedDaemon() *md.ManagedDaemon {
```

Comment:
Can this be defined in daemon_manager.go? That would then avoid duplication across the linux and windows implementations. Also, can the lines from 56-66 be moved completely there?